boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -209,6 +209,11 @@ endif
 
 LDELFFLAGS = -r -e main --gc-sections
 LDELFFLAGS += -T $(call CONVERT_PATH,$(BOARD_DIR)$(DELIM)scripts$(DELIM)gnu-elf.ld)
+LDFLAGS += -Wl,--gc-sections
+
+ifeq ($(CONFIG_HOST_MACOS),y)
+  LDFLAGS += -Wl,-dead_strip
+endif
```

Comment:
Probably gc-sections should be in else case

Reply:
Done